packages/node/src/behavior/system/software-update/OtaAnnouncements.ts

```diff
@@ -0,0 +1,240 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { CommissioningClient } from "#behavior/system/commissioning/CommissioningClient.js";
+import { NetworkClient } from "#behavior/system/network/NetworkClient.js";
+import { BasicInformationClient } from "#behaviors/basic-information";
+import { OtaSoftwareUpdateRequestorClient } from "#behaviors/ota-software-update-requestor";
+import { OtaSoftwareUpdateRequestor } from "#clusters/ota-software-update-requestor";
+import { Endpoint } from "#endpoint/Endpoint.js";
+import { Duration, Hours, isDeepEqual, Logger, MatterError, Millis, Minutes, Seconds, Time, Timer } from "#general";
+import type { ClientNode } from "#node/ClientNode.js";
+import { Node } from "#node/Node.js";
+import type { ServerNode } from "#node/ServerNode.js";
+import { Fabric, PeerAddress, Write } from "#protocol";
+import { EndpointNumber } from "#types";
+
+const logger = new Logger("OTAAnnouncements");
+
+export class OtaAnnouncements {
+    #announcementQueue = new Array<PeerAddress>();
+    #announcementTimer: Timer;
```

Comment:
Is there any reason to do this with bare timers rather than a loop in an async function with (abortable) sleeps?  I find the logic a lot easier to follow that way

Reply:
mainly because was convenient to reuse the timer for an initial delay started in constructor and then re-set for the others ... and even with a cancelable promise I need to track and handle it in several places. Was not that much of a benefit for me tbh